### PR TITLE
Update fee providing for bitcoin cash, zcash, and litecoin

### PR DESCRIPTION
**Pull request by cpacia**
Remove the dependency on cpacia/BitcoinCash-Wallet by moving the packages that were used into this repo.

Also switches zcash and litecoin to use the exchange rate based fee provider. 

And lowers the default fees to .1 cents, 1 cent, and 5 cents.

### CODE REVIEW

File: util/fees_test.go [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+func (m *mockExchangeRate) UnitsPerCoin() int {
+	return 0
+}
+
+func TestFeeProvider_GetFeePerByte(t *testing.T) {
+	fp := NewFeeProvider(2000, 360, 320, 280, &mockExchangeRate{})
+
+	// Test using exchange rates
+	if fp.GetFeePerByte(wallet.PRIOIRTY) != 50 {
```
Comment (placer14) [inline util/fees_test.go]: I don't see how the 360 amount turns into 50 given the mockExchangeRateProvider doesn't have any rates for USD.

**Comment by placer14** [comment]
Note that travis is also red.

File: util/fees_test.go (hunk 1 above)
Comment (cpacia) [reply util/fees_test.go]: It was in the `returnedRate` var. I've move the var inside the struct.

File: util/fees.go [hunk 2]
```diff
@@ -2,35 +2,105 @@ package util
...
+		target = EconomicTarget
 	case wallet.FEE_BUMP:
-		return fp.priorityFee * 2
+		target = PriorityTarget * 2
 	default:
-		return fp.normalFee
+		target = NormalTarget
 	}
+
+	feePerByte := (((float64(target) / 100) / rate) * 100000000) / 226
```
[the remark this answers is not in the document]
Comment (cpacia) [reply util/fees.go]: Average transaction size. I've moved it into a const

File: util/fees_test.go (hunk 1 above)
Comment (placer14) [reply util/fees_test.go]: It looks as though the mockExchangeRate doesn't have any rates... which leads me to think it's using the default fee?

**Pull request merged**